app/controllers/api_controller/providers.rb

```diff
@@ -5,9 +5,14 @@ module Providers
     CREDENTIALS_ATTR  = "credentials"
     AUTH_TYPE_ATTR    = "auth_type"
     DEFAULT_AUTH_TYPE = "default"
-    ENDPOINT_ATTRS    = %w(hostname ipaddress port)
+    ENDPOINT_ATTRS    = %w(hostname ipaddress port connection_configurations)
     RESTRICTED_ATTRS  = [TYPE_ATTR, CREDENTIALS_ATTR, ZONE_ATTR, "zone_id"]
 
+    def show_providers
+      @req[:additional_attributes] = ["endpoints"]
```

Comment:
You sure you always want to add this resource?
You can also do something like 
```
if @req[:c_id] == "endpoints" do
  @req[:additional_attributes] = ["endpoints"]
end
show_generic(:providers)
```
Then using api/providers/endpoints will bring back the provider plus the endpoints.